fgpyo/io/io.py

```diff
@@ -0,0 +1,178 @@
+"""
+IO
+-------
+Module for reading and writing files.
+
+The functions in this module make is easy to:
+    check if a file exists and is writeable
+    check if a file and its parent directories exist and are writeable
+    check if a file exists and is readable
+    check if a path exists and is a directory
+    open an appropriate reader or writer based on the file extension
+    writitng items to a file
+    reading lines from a file
+~~~~~~~~
+Examples:
+.. code-block:: python
+   >>> import fgpyo.io.io as IO
+   >>> from pathlib import Path
+   Assert that paths exist and are readable
+   >>> path_flath: Path = Path("example.txt")
+   >>> path_compressed: Path = Path("example.txt.gz")
+   >>> IO.paths_are_readable(path_flat)
+   AssertionError: Cannot read non-existent path: example.txt
+   >>> IO.paths_are_readable(compressed_file)
+   AssertionError: Cannot read non-existent path: example.txt.gz
+   Write to and read from path
+   >>> write_lines(path = path_flat, lines_to_write=["flat file", 10])
+   >>> write_lines(path = path_compressed, lines_to_write=["gzip file", 10])
+   Read lines from paths into a list of strings
+   >>> read_lines(path = path_flat)
+   ['flat file', '10']
+   >>> read_lines(path = path_compressed)
+   ['gzip file', '10']
+"""
+
+import gzip
+import io
+import os
+from pathlib import Path
+from typing import IO
+from typing import Any
+from typing import Iterable
+from typing import Set
+from typing import TextIO
+from typing import Union
+from typing import cast
+
+COMPRESSED: Set[str] = {".gz", ".bgz"}
+
+
+def paths_are_readable(path: Path) -> None:
+    """Checks that file exists and returns True, else raises FileNotFoundError
+
+    Args:
+        paths: a list of one or more Paths to be investigated
+
+    Example:
+    _file_exists(path = Path("some_file.csv"))
+    """
+    assert path.exists() is True, f"Cannot read non-existent path: {path}"
```

Comment:
Apply here and below since we know the return types are boolean (and not list or something else "truthy")
1. Can get rid of `is True` everywhere
2. Can change `foo is False` to `not foo`

Reply:
Fixed